plugins/scaffolder-backend/src/schema/openapi.yaml

```diff
@@ -0,0 +1,1055 @@
+        state:
+          $ref: '#/components/schemas/JsonObject'
+      required:
+        - id
+        - spec
+        - status
+        - createdAt
+      description: SerializedTask
+    TaskEventType:
+      type: string
+      description: TaskEventType
+      enum:
+        - completion
+        - log
+        - cancelled
+        - recovered
+    TaskRecovery:
+      type: object
+      properties:
+        EXPERIMENTAL_strategy:
+          $ref: '#/components/schemas/TaskRecoverStrategy'
+      description: |-
+        When task didn't have a chance to complete due to system restart you can define the strategy what to do with such tasks,
+        by defining a strategy.
+
+        By default, it is none, what means to not recover but updating the status from 'processing' to 'failed'.
+    TaskRecoverStrategy:
+      type: string
+      description: |
+        - none: not recover, let the task be marked as failed
+        - startOver: do recover, start the execution of the task from the first step.
+      enum:
+        - none
+        - startOver
+    TaskSecrets:
+      type: object
+      properties:
+        backstageToken:
+          type: string
+      additionalProperties:
+        type: string
+      description: TaskSecrets
+    TaskSpec:
+      $ref: '#/components/schemas/TaskSpecV1beta3'
+    TaskSpecV1beta3:
+      type: object
+      properties:
+        apiVersion:
+          type: string
+          enum:
+            - "scaffolder.backstage.io/v1beta3"
+          description: The apiVersion string of the TaskSpec.
+        parameters:
+          $ref: '#/components/schemas/JsonObject'
+          description: |
+            This is a JSONSchema which is used to render a form in the frontend
+            to collect user input and validate it against that schema. This can then be used in the `steps` part below to template
+            variables passed from the user into each action in the template.
+        steps:
+          type: array
+          items:
+            $ref: '#/components/schemas/TaskStep'
+          description: |
+            A list of steps to be executed in sequence which are defined by the template. These steps are a list of the underlying
+            javascript action and some optional input parameters that may or may not have been collected from the end user.
+        output:
+          type: object
+          additionalProperties:
+            $ref: '#/components/schemas/JsonValue'
+          description: |
+            The output is an object where template authors can pull out information from template actions and return them in a known standard way.
+        templateInfo:
+          $ref: '#/components/schemas/TemplateInfo'
+          description: Some information about the template that is stored on the task spec.
+        user:
+          type: object
+          properties:
+            entity:
+              $ref: '../../../catalog-backend/src/schema/openapi.yaml#/components/schemas/UserEntity'
```

Comment:
To be honest I think it's best to keep this one opaque. As an adopter of Backstage you can extend the `UserEntity` schema anyway, so there's kinda no point aiming to model this imo.

Reply:
Fair point. I'll adjust here, and remove the addition from the Catalog OpenAPI schema.